src/test/java/io/github/spark_redshift_community/spark/redshift/InMemoryS3AFileSystem.java

```diff
@@ -199,18 +199,18 @@ private boolean isDir(Path f) throws IOException{
     public S3AFileStatus getFileStatus(Path f) throws IOException {
 
         if (!exists(f)) throw new FileNotFoundException();
-        FileStatus fileStatus = new FileStatus(
-            dataMap.get(toS3Key(f)).toByteArray().length, true, 1,
-            this.getDefaultBlockSize(), System.currentTimeMillis(), f
-        );
+        Tristate state = Tristate.fromBool(dataMap.tailMap(toS3Key(f)).size() == 1 && dataMap.containsKey(toS3Key(f)));
 
         if (isDir(f)) {
-            S3AFileStatus status = S3AFileStatus.fromFileStatus(
+            FileStatus fileStatus = new FileStatus(
+                dataMap.tailMap(toS3Key(f)).size(), true, 1,
+                this.getDefaultBlockSize(), System.currentTimeMillis(), f
+            );
+            return S3AFileStatus.fromFileStatus(
                 fileStatus, Tristate.fromBool(
                     dataMap.tailMap(toS3Key(f)).size() == 1 && dataMap.containsKey(toS3Key(f))
                 )
             );
```

Comment:
the `Tristate.fromBool(...)` is `state` right?

Reply:
I think so. I kept the state same in the original code. Also, verified with one of the unit tests.
I debugged the code to check the values for empty-path and non-empty path.
IIRC, I think dataMap.tailMap(toS3Key(f)).size() is 1 is probably only counting self (i.e. empty-dir). I can double check again, if this seems doubtful.